src/ensembl/src/content/app/custom-download/containers/content/preview-card/previewCardHelper.ts

```diff
@@ -0,0 +1,111 @@
+import JSONValue from 'src/shared/types/JSON';
+
+import {
+  flattenObject,
+  getProcessedAttributes
+} from 'src/content/app/custom-download/containers/content/customDownloadContentHelper';
+
+const formatResponseToArray = (responseData: any): any => {
+  const preResult: any = [];
+
+  const responseArray = flattenObject(responseData);
+  Object.keys(responseArray)
+    .sort()
+    .forEach((key) => {
+      const keySplit: string[] = key.split('.');
+
+      let topID = keySplit[0];
+      topID += keySplit[2] ? keySplit[2] : '0';
+      topID += keySplit[4] ? keySplit[4] : '0';
+
+      let id = keySplit[1];
+      if (keySplit[3]) {
+        id = `${keySplit[1]}.${keySplit[3]}`;
+      }
+
+      if (!preResult[topID]) {
+        preResult[topID] = {
+          ...preResult[`${keySplit[0]}00`]
+        };
+      }
+      preResult[topID][id] = responseArray[key];
+    });
+
+  return preResult;
```

Comment:
I am seeing a preResult that is and array with over 7000 elements, most of them empty:

![image](https://user-images.githubusercontent.com/6834224/66832118-1da12400-ef51-11e9-896e-ca7ce7150270.png)

![image](https://user-images.githubusercontent.com/6834224/66832186-45908780-ef51-11e9-874e-cac1fa2e250f.png)

Is this intended?

Also, why `pre`Result, why not result?

Reply:
I've renamed it to `result`.

Yes, it is intended. I've just implemented this as a temporary solution as all these will be done in the backend.

Reply:
Wouldn't using object instead of an array (and returning `Object.values` of that object) be a more natural choice? Is there any specific array-like behaviour that you needed from the result while building it?

Reply:
Ok. It's just that such a humungous and mostly empty array looks terrifying.